Autocomplete

CedricBabeix opened the pull request:
In android 5.0.2, the URL autocompletion wasn't working properly, every
text appened to the URL was unselected before the user enters a new
character.
"a" was autocompleted to "abcnews.com", the cursor being between "a" and
"b", if the user then typed "b", the resulting URL was "abbcnews.com".

Commit:
Autocomplete

In android 5.0.2, the URL autocompletion wasn't working properly, every
text appened to the URL was unselected before the user enters a new
character.
"a" was autocompleted to "abcnews.com", the cursor being between "a" and
"b", if the user then typed "b", the resulting URL was "abbcnews.com".

SergeyZhukovsky commented:
I have a device with exactly the same version and don't have any problems with that. I've tried it on several devices and that condition if(difference == 1) {  is always false for me. @CedricBabeix do you have any other steps how to reproduce it?
Thanks.

CedricBabeix commented:
Hello,

I tried to reproduce this bug on several virtual devices after reading your comment and I finally found out where it comes from : the [ZenUI keyboard](https://play.google.com/store/apps/details?id=com.asus.ime).

I got an [Asus ZenPad 80](https://www.asus.com/Tablets/ASUS_ZenPad_80_Z380C/), running Android 5.0.2 with the ZenUI user interface and all it's related apps.

I was able to reproduce the bug using the following procedure : 

- Create a virtual device on your Android emulator.
- Configure your virtual device to use the virtual keyboard, not your computer's keyboard.
- Install the latest ZenUI keyboard : [www.apkmirror.com](http://www.apkmirror.com/apk/zenui-asus-computer-inc/asus-keyboard/asus-keyboard-1-7-8-14_160929-release/asus-zenui-keyboard-1-7-8-14_160929-android-apk-download/).
- Configure your virtual device to use the ZenUI keyboard.
    _Settings_ > _Language & input_ > _Current keyboard_ > _Choose keyboards_ >  _ZenUI Keyboard_
    In _Settings_ > _Language & input_, tap again on _Current keyboard_ and select the _ZenUI Keyboard_ option.
- Install Brave / Link Bubble.
- Open Brave / Link Bubble.
- Delete the current URL if any.
- Type in "a", the URL will be autocompleted to "abcnews.go.com", the cursor being between "a" and "b".
- Type in "b", the previously appended string "bcnews.go.com" should be deleted and replaced by "b", instead "b" is inserted between "a" and "bcnews.go.com", resulting in the URL "abbcnews.go.com".

I reproduced the bug on two virtual devices, one running on Android 5.0.0 and the other running on Android 6.0.0.

Cédric

Commit:
Changed "previousAppenedString" to "previousAppendedString"

SergeyZhukovsky commented:
Thank for the detailed steps @CedricBabeix. I was able to replicate it with emulator and ZenUI keyboard. It seems like the fix is fine to me. Could you only correct that 
private String previousMetUrl;
private String previousAppenedString;
to make them the same style as we have in Link Bubble
private String mPreviousMetUrl;
private String mPreviousAppenedString;

I'm going to land that after that fix.
Thanks.

Commit:
Text formatting

Changed "previousMetUrl" to "mPreviousMetUrl" and "previousAppendedUrl" to "mPreviousAppendedUrl".

CedricBabeix commented:
You're welcome.
Correction is done.

SergeyZhukovsky commented:
Thanks @CedricBabeix, nice catch!
